### PR TITLE
use cleaned repository substitutions everywhere

### PR DESCRIPTION
**Pull request by evgeni**
before we did use them for searching, but not in the resulting error
message, which can be confusing

### CODE REVIEW

File: plugins/modules/katello_repository_set.py [hunk 1]
```diff
@@ -152,7 +152,6 @@
 def get_desired_repos(desired_substitutions, available_repos):
```
Comment (ekohl) [inline plugins/modules/katello_repository_set.py]: Should this now be `return itertools.chain(filter(...) for sub in desired_substitutions)`?

File: plugins/modules/katello_repository_set.py [hunk 2]
```diff
@@ -152,7 +152,6 @@
...
     desired_repos = []
     for sub in desired_substitutions:
-        sub = {k: v for (k, v) in sub.items() if v is not None}
         desired_repos += filter(lambda available: available['substitutions'] == sub, available_repos)
```
Comment (ekohl) [inline plugins/modules/katello_repository_set.py]: Does this work in Python 3? Isn't this a `list += generator`?

Comment (evgeni) [reply plugins/modules/katello_repository_set.py]: It does work :)

File: plugins/modules/katello_repository_set.py (hunk 1 above)
Comment (evgeni) [reply plugins/modules/katello_repository_set.py]: maybe? ;)

Comment (evgeni) [reply plugins/modules/katello_repository_set.py]: `TypeError: 'filter' object is not subscriptable` and I am not in the mood to understand why ;)

**Pull request merged**